client/client.go

```diff
@@ -140,27 +139,56 @@ func (c *genericClient) getResponseTo(msg *pb.Message) (*pb.Message, error) {
 	return c.receive()
 }
 
-// openStream opens the gRPC communication stream with the server prior to actual execution of
-// the protocol client.
+// openStream is a generic function for opening a pb.ClientStream.
+// A pb.ClientStream is generated as a result of the function call of the form
+// stream, err := grpcClient.streamGenFunc(context.Background()). As we have different
+// grpcClients (each generated from its own RPC service), each has its own streamGenFunc(s)
+// (generated from the appropriate RPC within the service), it is the caller's responsibility
+// to provide appropriate grpcClient and streamGenFunc.
 // This function has to be called explicitly at the beginning of the protocol execution function.
-func (c *genericClient) openStream() error {
-	stream, err := c.protocolClient.Run(context.Background())
+func (c *genericClient) openStream(grpcClient interface{}, streamGenFunc string) error {
+	// Create structs compatible with reflect package
+	client := reflect.ValueOf(grpcClient)                            // we want to call streamGenFunc on this struct
+	params := []reflect.Value{reflect.ValueOf(context.Background())} // we want to pass these params to streamGenFunc
```

Comment:
`params` indeed represents a set of arguments to be passed to `streamGenFunc` (`streamGenFunc` is actually a general representation of RPCs like `Run`, `GenerateNym`, `GenerateNym_EC`,...). 

However, `params` is not an empty set of arguments; on L152, you will notice that we're actually using reflection to construct a slice of `reflect.Value`s that has a single element, which is `reflect.Value` representation of `context.Background()`. The result of `context.Background()` call is a value that we pass to all of the above mentioned `streamGenFunc`s  (see L161)  - we always use the same value regardless of the actual `streamGenFunc` being called.

So, `openStream` doesn't have a `params` argument because we don't expect it to be different between invocations of different `streamGenFunc`s - all the `streamGenFunc`s expect a `context.Context` argument, which can indeed be something other than the result of `context.Background()` call, but at this point, we never used anything else, so the same value is used during invocation of the `streamGenFunc`s.